tests/unit/Core/Checkout/Cart/Price/CashRoundingTest.php

```diff
@@ -69,6 +69,25 @@ public function testSweden(float $price, float $expected): void
         static::assertEquals($expected, $actual);
     }
 
+    public function testNegativeZeroHandling(): void
+    {
+        $service = new CashRounding();
+        $config = new CashRoundingConfig(2, 0.01, true);
+
+        // Test case mimicking the scenario with line item 125, delivery shipping cost of 4.9 and line item -129.9
+        $sum = 125 + 4.9 - 129.9; // Expected to be 0 but could be -0 due to floating point arithmetic
+
+        $result = $service->mathRound($sum, $config);
+
+        // Convert to string to check if it's positive zero (should not start with '-')
+        $resultAsString = sprintf('%.1f', $result);
+        static::assertEquals('0.0', $resultAsString);
```

Comment:
```suggestion
        static::assertSame('0.0', $resultAsString);
```

please use `assertSame` where possible instead, as it is type safe